hawkmoth/parser.py

```diff
@@ -255,6 +256,11 @@ def parse(filename, **options):
                      TranslationUnit.PARSE_DETAILED_PROCESSING_RECORD |
                      TranslationUnit.PARSE_SKIP_FUNCTION_BODIES)
 
+    for diag in tu.diagnostics:
+        if diag.severity >= 2:
+            parser_errs.append('S {}, {}, {}'.format(diag.severity,
+                                                     diag.location,
+                                                     diag.spelling))
```

Comment:
Don't like this so much. I prefer to pass the information as a tuple as shown in my example. That way we can print different severity messages as warnings or errors. Note that this module should stand on its own, maybe we want to use the error stuff differently somewhere else without regexing a string.

On another note, I'd prefer `diag.location` to be split in `.file.name` and `.line`. That's more general and consistent once we add other non Clang related warnings and errors.